helpers/jsonParseSafe.js

```diff
@@ -10,7 +10,10 @@ function factory(globals) {
       }
       return JSON.parse(value);
     } catch (err) {
-      return options.inverse(this);
+      if (options.fn) { // If function block {{#JSONparseSafe "{}"}} escape here on excaption.
+        return options.inverse(this);
+      }
+      return {}; // Escape hatch to avoid crash due to inline malformed JSON input.
```

Comment:
```suggestion
      return undefined; // Escape hatch to avoid crash due to inline malformed JSON input.
```
It's probably best to return `undefined` when no valid object or value could be parsed.

Reply:
At this moment I don't have any outright objections to the response being `undefined` but I think it should be noted for future edits of the inline spec, or new helper developments.

The current block spec is as follows:
```handlebars
{{log "--------------"}}
{{#JSONparseSafe "{"}}
    {{log "safe block"}}
    {{log this}} {{!-- this is undefined if used as inline --}}
{{else}}
    {{log "else block"}}
{{/JSONparseSafe}}
{{log "--------------"}}
```

In the above example, `JSONparseSafe` will enter the false block.
Therefore I would expect that the following is valid code for use when calling as a inline:

```handlebars
{{#with (extend (JSONparseSafe 'BAD JSON') (JSONparseSafe '{"test":true}'))}}
    {{log this}} {{!-- logs { test: true } with current & original proposal --}}
{{/with}}
```

Reply:
`(extend (JSONparseSafe 'BAD JSON') (JSONparseSafe '{"test":true}'))` should work regardless, since `extend` creates its own result object and skips non-object arguments (https://github.com/bigcommerce/paper-handlebars/blob/4fbc7b7efd6f6904753daa376a7c08a6d5055a0d/helpers/3p/object.js#L32-L37).

Reply:
true, I think this is ready to merge / argue about internally at BC.
Good luck getting this through to production 😄  
- Jordan